src/common/auth.rs

```diff
@@ -24,22 +30,70 @@ impl AuthKeys {
             (read_write, read_only) => Some(Self {
                 read_write,
                 read_only,
+                jwt_parser: service_config
+                    .api_key
+                    .as_ref()
+                    .map(|secret| JwtParser::new(secret)),
             }),
         }
     }
 
+    /// Validate that the specified request is allowed for given keys.
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(None)` if the request is allowed through the API key.
+    /// - `Ok(Some(claims))` if the request is allowed through the JWT token.
+    /// - `Err(description)` if the request is not allowed.
+    pub fn validate_request<'a>(
+        &self,
+        get_header: impl Fn(&'a str) -> Option<&'a str>,
+        is_read_only: bool,
+    ) -> Result<Option<Claims>, &'static str> {
+        if let Some(api_key) = get_header("api-key") {
+            if self.can_write(api_key) || (is_read_only && self.can_read(api_key)) {
+                return Ok(None);
+            } else {
+                return Err("Invalid API key");
+            }
+        }
+
+        if let Some(bearer_token) =
+            get_header("authorization").and_then(|v| v.strip_prefix("Bearer "))
```

Comment:
Can we assume that `api-key` and `authorization: Bearer` are the same thing? I suppose it currently works like this